matrix-react-sdk/pipeline.yaml

```diff
@@ -18,7 +18,11 @@ steps:
       - "echo '+++ Lint'"
       - "yarn lint:types"
       - "echo '--- Switch js-sdk to release mode'"
-      - "./scripts/ci/js-sdk-to-release.sh"
+      - "./scripts/ci/js-sdk-to-release.js"
+      - "pushd node_modules/matrix-js-sdk"
+      - "yarn run build:compile"
+      - "yarn run build:types"
+      - "popd"
```

Comment:
wouldn't it be more appropriate to just change `scripts/ci/js-sdk-to-release.sh` (or make another script)?

Tbh this whole thing (from `install-deps` through to `yarn lint:types` should be in a script in the repo. Having 59 steps in the build pipeline increases the coupling between the `matrix-react-sdk` repo and the `pipelines` repo, making it harder to change things in future (because both repos have to updated at once).